packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -3,82 +3,72 @@ use chrono::{DateTime, Utc};
+        query::{EntityQuery, EntityVersion},
+        AsClient, PostgresStore, QueryError,
+    },
 };
 
-#[async_trait]
-impl<C: AsClient> crud::Read<'_, EntityId, PersistedEntity> for PostgresStore<C> {
-    type Output = PersistedEntity;
-
-    async fn get(&self, identifier: EntityId) -> Result<Self::Output, QueryError> {
-        let row = self
-            .as_client()
-            .query_one(
-                r#"
-                SELECT properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+async fn by_id_by_latest_version(
+    client: &(impl GenericClient + Sync),
+    entity_id: EntityId,
+) -> Result<RowStream, QueryError> {
+    client
+        .query_raw(
+            r#"
+            SELECT properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+            FROM entities
+            INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
+            WHERE entity_id = $1 AND entities.version = (
+                SELECT MAX("version")
                 FROM entities
-                INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
-                WHERE entity_id = $1 AND entities.version = (
-                    SELECT MAX("version")
-                    FROM entities
-                    WHERE entity_id = $1
-                );
-                "#,
-                &[&identifier],
-            )
-            .await
-            .into_report()
-            .change_context(QueryError)
-            .attach_printable(identifier)?;
-
-        let entity = serde_json::from_value(row.get(0))
-            .into_report()
-            .change_context(QueryError)?;
-
-        let entity_id: EntityId = row.get(1);
-        let entity_version: DateTime<Utc> = row.get(2);
-        let type_base_uri: String = row.get(3);
-        let type_version: i64 = row.get(4);
-        let created_by: AccountId = row.get(5);
+                WHERE entity_id = $1
+            );
+            "#,
+            &[&entity_id],
+        )
+        .await
+        .into_report()
+        .change_context(QueryError)
+}
 
-        let type_versioned_uri = VersionedUri::new(type_base_uri, type_version as u32);
-        Ok(PersistedEntity::new(
-            entity,
-            entity_id,
-            entity_version,
-            type_versioned_uri,
-            created_by,
-        ))
-    }
+async fn by_latest_version(client: &(impl GenericClient + Sync)) -> Result<RowStream, QueryError> {
+    client
+        .query_raw(
+            r#"
+            SELECT DISTINCT ON(entity_id) properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+            FROM entities
+            INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
+            ORDER BY entity_id, entities.version DESC;
+            "#,
+            // Requires a concrete type, which implements `IntoIterator<Item = impl BorrowToSql>`
+            [] as [&(dyn ToSql + Sync); 0],
```

Comment:
Couldn't we use the `parameter_list` utility function across `knowledge` and `ontology`? So we can replace this as well

Reply:
Applied in eefbbb505466911cf5d6ff78dd0ac2e8945eb546